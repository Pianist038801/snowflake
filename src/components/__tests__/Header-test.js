'use strict';

jest.autoMockOff();

import React, { View } from 'react-native';
import utils from 'react-addons-test-utils';

jest.dontMock('../Header');
var Header = require('../Header');

describe('Header', () => {
  let header;
  
  function renderHeader(props) {
    const renderer = utils.createRenderer();
    renderer.render(<Header {...props}/>);
    const output = renderer.getRenderOutput();

    return {
      props,
      output,
      renderer
    };
  }

  it('should be display empty text when not fetching', () => {
    const buttonProps = {
      isFetching: false
    };
    header = renderHeader(buttonProps);
    const {output} = header;
    expect(output.type).toEqual(View);
    expect(output.props.children[1].props.children).toEqual(' ');
  });
  
  it('should be display spinner when fetching', () => {
    const buttonProps = {
      isFetching: true
    };
    header = renderHeader(buttonProps);
    const {output} = header;
    expect(output.type).toEqual(View);
    expect(output.props.children[1].type.displayName).toEqual('GiftedSpinner');
  });

});//describe Header
