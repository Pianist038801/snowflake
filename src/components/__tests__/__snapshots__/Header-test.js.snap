exports[`Header should be display empty text when not fetching 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "justifyContent": "center",
        "marginTop": 10
      }
    }>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function bound touchableHandleResponderGrant]}
      onResponderMove={[Function bound touchableHandleResponderMove]}
      onResponderRelease={[Function bound touchableHandleResponderRelease]}
      onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
      onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
      onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent"
          },
          undefined
        ]
      }
      testID={undefined}>
      <Image
        source={
          Object {
            "uri": "http://i.imgur.com/da4G0Io.png"
          }
        }
        style={
          Object {
            "height": 50,
            "width": 50
          }
        } />
    </View>
    <Text>
       
    </Text>
  </View>
  <Text>
     
  </Text>
</View>
`;

exports[`Header should be display spinner when fetching 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "justifyContent": "center",
        "marginTop": 10
      }
    }>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function bound touchableHandleResponderGrant]}
      onResponderMove={[Function bound touchableHandleResponderMove]}
      onResponderRelease={[Function bound touchableHandleResponderRelease]}
      onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
      onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
      onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent"
          },
          undefined
        ]
      }
      testID={undefined}>
      <Image
        source={
          Object {
            "uri": "http://i.imgur.com/da4G0Io.png"
          }
        }
        style={
          Object {
            "height": 50,
            "width": 50
          }
        } />
    </View>
    <ActivityIndicator
      animating={true}
      size="large" />
  </View>
  <Text>
     
  </Text>
</View>
`;
